p2p/transport/webrtc/state.go

```diff
@@ -0,0 +1,59 @@
+package libp2pwebrtc
+
+import (
+	pb "github.com/libp2p/go-libp2p/p2p/transport/webrtc/pb"
+)
+
+type channelState uint8
+
+const (
+	stateOpen channelState = iota
+	stateReadClosed
+	stateWriteClosed
+	stateClosed
+)
+
+func (state channelState) handleIncomingFlag(flag pb.Message_Flag) channelState {
+	if state == stateClosed {
+		return state
+	}
+	switch flag {
+	case pb.Message_FIN:
+		if state == stateWriteClosed {
+			return stateClosed
+		}
+		return stateReadClosed
+
+	case pb.Message_STOP_SENDING:
+		if state == stateReadClosed {
+			return stateClosed
+		}
+		return stateWriteClosed
+	case pb.Message_RESET:
+		return stateClosed
+
```

Comment:
is it okay not to handle default case? Given Go doesn't give guarantees here

Reply:
Not sure if it's ok, but if it is, a comment explain why it's ok would be appreciated.